### PR TITLE
Remove a 'using' template alias.

### LINKED ISSUE #2772: C++11 failure when only -std=c++0x is available
I get a failure of the following form with GCC 4.6.3:
```
.../dealii/include/deal.II/base/quadrature_point_data.h:66:25: error: expected unqualified-id before ‘using’
```
This happens with the following set of lines from `include/deal.II/base/quadrature_point_data.h`
```
template <typename CellIteratorType, typename DataType>
class CellDataStorage : public Subscriptor
{
public:
  /**
   * A typedef for a vector of shared pointers with arbitrary data type @p T .
   */
  template <typename T> using CellVector = std::vector<std::shared_ptr<T> >;
```
The problem appears to be that GCC 4.6.3 has a flag `-std=c++0x` that makes the compiler accept *some* parts of C++11, but apparently not all -- in particular not template aliases (using) statements.

How do we address this? Only switch on C++11 mode if a compiler actually understands C++11, not just has a C++0x flag?

### PR DESCRIPTION
**Pull request by drwells**
This is a C++11 feature that is not compatible with some older compilers that we would like to support (e.g., GCC 4.6 on Ubuntu 12.04).

The code is a bit uglier, but using auto helps.

@jppelteret @davydden Could you take a look at this and see if it breaks your code? All of the tests currently pass.

Should close #2772.

### CODE REVIEW

File: include/deal.II/base/quadrature_point_data.h [hunk 1]
```diff
@@ -520,7 +515,7 @@ void CellDataStorage<CellIteratorType,DataType>::initialize(const CellIteratorTy
       map[cell] = std::vector<std::shared_ptr<DataType> >(n_q_points);
       // we need to initialise one-by-one as the std::vector<>(q, T())
       // will end with a single same T object stored in each element of the vector:
-      typename std::map<CellIteratorType,CellVector<DataType> >::iterator it = map.find(cell);
+      auto it = map.find(cell);
```
Comment (davydden) [inline include/deal.II/base/quadrature_point_data.h]: is `auto` fine on that old gcc 4.6.3?

**Comment by davydden** [comment]
LGTM :+1:

File: include/deal.II/base/quadrature_point_data.h (hunk 1 above)
Comment (drwells) [inline include/deal.II/base/quadrature_point_data.h]: LinearOperator uses `auto` so I think that that should be okay.

Comment (davydden) [inline include/deal.II/base/quadrature_point_data.h]: ok. That makes code much more concise here :+1:

**Comment by jppelteret** [comment]
Thanks @drwells, all of the changes look alright to me as well. Its unfortunate that template aliases don't have universal support across the compilers that deal.II is tested against :-(

**Pull request merged**